packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -40,6 +63,7 @@ export interface PreferencesState extends BaseState {
   };
   showTestNetworks: boolean;
   isIpfsGatewayEnabled: boolean;
+  showIncomingTransactions: { [chainId: string]: boolean };
```

Comment:
We can narrow the index signature of the object so that it only accepts hex ids for etherscan supported chains specified in `constants.ts`.

```suggestion
  showIncomingTransactions: { [key in EtherscanSupportedHexChainId]: boolean };
```